libs/domains/services/data-access/src/lib/utils/refacto-service-payload.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  type ApplicationEditRequest,
+  type ApplicationGitRepositoryRequest,
+  DatabaseModeEnum,
+  type HelmRequest,
+  type HelmResponseAllOfSourceOneOf,
+  type HelmResponseAllOfSourceOneOf1,
+  type ServiceStorageStorageInner,
+} from 'qovery-typescript-axios'
+import { match } from 'ts-pattern'
+import { isJobGitSource } from '@qovery/shared/enums'
+import { type Application, type Container, type Database, type Helm, type Job } from '../domains-services-data-access'
+
+export function refactoApplication(application: Application) {
```

Comment:
It's a copy of `refacto-payload.ts` with new interfaces, I tried only `refactoHelm`, we need to clean other functions when we migrate all editing to React Query

Reply:
We can discuss it live but IMO those "refacto" functions shouldn't be necessary

Reply:
Already discussed it's needed because we can't pass some fields like `id`, `created_at`, `updated_at`, etc.

I created a Notion to list some improvement we can have on the new API: https://www.notion.so/qovery/API-improvements-b54ba305c2ee4e549eb002278c532c7f